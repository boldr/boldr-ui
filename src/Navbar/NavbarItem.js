// @flow
import * as React from 'react';
import classNames from 'classnames';
// import NavLink from 'react-router-dom/NavLink';
import { StyleClasses } from '../theme/styleClasses';
import { getActiveModifiers, removeActiveModifiers, createWrappedComponent } from '../util/boldrui';
import { getDomSafeProps } from '../util/helpers';

export type Props = {
  tag?: string,
  isHoverable?: boolean,
  hasDropdown?: boolean,
  render?: Function,
  title?: string,
  className?: string,
  href?: string,
};

const BASE_ELEMENT = StyleClasses.NAVBAR_ITEM;
export function NavbarItem({ tag = 'div', render, isHoverable, hasDropdown, ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      'is-hoverable': isHoverable,
      'has-dropdown': hasDropdown,
      ...getActiveModifiers(props),
    },
    props.className,
  );

  const HTMLProps = getDomSafeProps(props, removeActiveModifiers);

  if (render) {
    return render({ ...HTMLProps, className });
  }
  if (props.href) {
    return (
      <a className={className} href={props.href} {...HTMLProps}>
        {props.title}
      </a>
    );
  }
  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(NavbarItem);
