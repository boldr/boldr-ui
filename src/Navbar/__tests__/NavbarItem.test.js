import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarItem } from '../NavbarItem';

describe('<NavbarItem />', () => {
  it('should render a custom component with NavbarItem props', () => {
    const customComponent = props => (
      <div>
        My Button <a {...props} />
      </div>
    );

    const customComponentRendered = (
      <div>
        My Button <a href="#" className="boldr-navbar__item" />
      </div>
    );

    const component = shallow(<NavbarItem href="#" render={customComponent} />);
    expect(component.contains(customComponentRendered)).toBe(true);
  });

  // it('should render an anchor with .boldr-navbar__item', () => {
  //   const component = shallow(<NavbarItem href="#">My NavbarItem</NavbarItem>);
  //   expect(
  //     component.contains(
  //       <a href="#" className="boldr-navbar__item">
  //         My NavbarItem
  //       </a>,
  //     ),
  //   ).toBe(true);
  // });

  it('should render a button with .boldr-navbar__item', () => {
    const component = shallow(<NavbarItem tag="button">My NavbarItem</NavbarItem>);
    expect(component.contains(<button className="boldr-navbar__item">My NavbarItem</button>)).toBe(
      true,
    );
  });

  it('should render a div with .boldr-navbar__item', () => {
    const component = shallow(<NavbarItem>My NavbarItem</NavbarItem>);
    expect(component.contains(<div className="boldr-navbar__item">My NavbarItem</div>)).toBe(true);
  });

  it('should render a div with .boldr-navbar__item and modifiers', () => {
    const component = shallow(<NavbarItem isActive isHoverable hasDropdown />);
    expect(component.is('div')).toBe(true);
    expect(component.hasClass('boldr-navbar__item')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
    expect(component.hasClass('is-hoverable')).toBe(true);
    expect(component.hasClass('has-dropdown')).toBe(true);
  });

  it('should render a div with .boldr-navbar__item, modifiers and custom classNames', () => {
    const component = shallow(<NavbarItem isActive className="custom" />);
    expect(component.is('div')).toBe(true);
    expect(component.hasClass('boldr-navbar__item')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });

  it('should render a div with .boldr-navbar__item and custom classNames', () => {
    const component = shallow(<NavbarItem className="custom" />);
    expect(component.is('div')).toBe(true);
    expect(component.hasClass('boldr-navbar__item')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarItem href="#">My NavbarItem</NavbarItem>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
