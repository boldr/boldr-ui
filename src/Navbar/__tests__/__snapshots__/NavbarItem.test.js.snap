// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarItem /> should match the snapshot 1`] = `
<div
  className="boldr-navbar__item"
  href="#"
>
  My NavbarItem
</div>
`;
